modules/siddhi-extensions/regex/src/test/java/org/wso2/siddhi/extension/regex/RegexPersistanceTestCase.java

```diff
@@ -62,200 +67,217 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
                         Assert.assertEquals(true, inEvent.getData(1));
                     }
                     if (count == 2) {
-                        Assert.assertEquals(true, inEvent.getData(1));
+                        Assert.assertEquals(false, inEvent.getData(1));
                     }
                     if (count == 3) {
-                        Assert.assertEquals(true, inEvent.getData(1));
+                        Assert.assertEquals(false, inEvent.getData(1));
                     }
                     if (count == 4) {
-                        Assert.assertEquals(false, inEvent.getData(1));
+                        Assert.assertEquals(true, inEvent.getData(1));
                     }
                     eventArrived = true;
                 }
             }
-        });
-
+        };
+        executionPlanRuntime.addCallback("query1", queryCallback);
         InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
         executionPlanRuntime.start();
-        inputHandler.send(new Object[]{"21 products are produced by WSO2 currently", 60.5f, "\\d\\d(.*)WSO2"});
-        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2(.*)middleware(.*)"});
-        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2(.*)middleware"});
+        inputHandler.send(new Object[]{"21 products are produced by WSO2 currently", 60.5f, "WSO2"});
+        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2"});
+        inputHandler.send(new Object[]{"WSO2 is situated in trace and its a middleware company", 60.5f, "WSO2"});
+
+        //persisting
+        Thread.sleep(500);
+        executionPlanRuntime.persist();
+        //restarting execution plan
+        executionPlanRuntime.shutdown();
+        inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+        //loading
+        executionPlanRuntime.restoreLastRevision();
         inputHandler.send(new Object[]{"21 products are produced by WSO2 currently", 60.5f, "\\d(.*)WSO22"});
-        Thread.sleep(100);
+        Thread.sleep(500);
```

Comment:
Use SiddhiTestHelper instead of sleeps. Refer https://github.com/wso2/siddhi/blob/3.1.x/modules/siddhi-core/src/test/java/org/wso2/siddhi/core/managment/PersistenceTestCase.java#L931